demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c

```diff
@@ -168,186 +589,245 @@ CK_RV xProvisionCertificate( CK_SESSION_HANDLE xSession,
                                                     xCertificateTemplate.xValue.ulValueLen,
                                                     pucDerObject,
                                                     &xDerLen );
-        }
-        else
-        {
-            xResult = CKR_DEVICE_MEMORY;
-        }
 
-        if( lConversionReturn == 0 )
-        {
-            /* Set the template pointers to refer to the DER converted objects. */
-            xCertificateTemplate.xValue.pValue = pucDerObject;
-            xCertificateTemplate.xValue.ulValueLen = xDerLen;
+            if( 0 != lConversionReturn )
+            {
+                xResult = CKR_ARGUMENTS_BAD;
+            }
         }
         else
         {
-            configPRINTF( ( "ERROR: Failed to provision certificate %d \r\n", lConversionReturn ) );
+            xResult = CKR_HOST_MEMORY;
         }
+    }
 
-        /* Create an object using the encoded client certificate. */
-        if( CKR_OK == xResult )
-        {
-            configPRINTF( ( "Write certificate...\r\n" ) );
+    if( xResult == CKR_OK )
+    {
+        /* Set the template pointers to refer to the DER converted objects. */
+        xCertificateTemplate.xValue.pValue = pucDerObject;
+        xCertificateTemplate.xValue.ulValueLen = xDerLen;
+    }
 
-            xResult = pxFunctionList->C_CreateObject( xSession,
-                                                      ( CK_ATTRIBUTE_PTR ) &xCertificateTemplate,
-                                                      sizeof( xCertificateTemplate ) / sizeof( CK_ATTRIBUTE ),
-                                                      pxObjectHandle );
-        }
+    /* Create an object using the encoded client certificate. */
+    if( xResult == CKR_OK )
+    {
+        configPRINTF( ( "Write code signing certificate...\r\n" ) );
```

Comment:
This is for sure handling the code-signing certificate in all cases?

Reply:
no it is not.  I have changed the comment to just say write certificate